src/button/icon-helper.tsx

```diff
@@ -18,6 +18,8 @@ export interface ButtonIconProps {
   iconSize?: IconProps.Size;
   variant?: string;
   iconClass?: string;
+  badge?: boolean;
+  badgeColor?: 'blue' | 'red';
```

Comment:
Interesting things happen here. 

The blue color used in the top navigation for the same reason, highlight active notifications. Do we really need separate colors in different places? Also, even if different colors, how about using visual contexts to modify a single design token for both?

Reply:
We had a design team discussion about the blue vs. red color and decided to leave both for now. We may want to keep both variants in the future, but how would I use visual contexts in this case?

Reply:
We can define a design token, `colorBackgroundNotificationBadge`, for example (also to address this comment https://github.com/cloudscape-design/components/pull/1416#discussion_r1287022139), default to red and then redefine it to blue in this context only: https://github.com/cloudscape-design/components/blob/main/style-dictionary/classic/contexts/top-navigation.ts

Reply:
So I ended up realizing there are already design tokens used in the badge component itself - `awsui.$color-background-notification-red;` and `awsui.$color-background-notification-blue;` which I updated in the PR. What do you think of using those? They convey the same message without having to add anything additional.

Reply:
I am not sure about exposing blue/red choice in the API. For me this choice is contextual, based on background (thus perfectly fits visual contexts use case).

Also, some may argue that badge should use "color-text" tokens and not "color-background", because there is no text inside the badge.

My personal opinion is still the same – "icon-badge" is a separate feature than the "badge" component and should use its own token. But we can discuss with our squad if there are different opinions

Reply:
So I brought this back up with the design team today and we decided to just make the notification badge red everywhere. So this becomes a non-issue anyway!